### ISSUE TITLE
Backups not incremental for chrooted domains

### ISSUE THREAD

**Issue by ghost** [virtualmin/virtualmin-gpl#758]
Hi,

Since the release of Virtualmin version 7.9.0, incremental backups for chrooted domains have been observed to be the same size as full backups. Further information can be found in the following two forum threads:

[Backups not incremental for chrooted domains](https://forum.virtualmin.com/t/backups-not-incremental-for-chrooted-domains/124900)
[Incremental backup same size as full backup](https://forum.virtualmin.com/t/incremental-backup-same-size-of-full-backup/124790)

Thank you.

**Comment by jcameron** [comment]
Yes, it looks like the issue is this chmod command : https://github.com/virtualmin/virtualmin-gpl/blob/master/jailkit-lib.pl#L299

@iliajie you added this code recently .... do we really need that chmod?

**Comment by iliajie** [comment]
> Yes, it looks like the issue is this chmod command : https://github.com/virtualmin/virtualmin-gpl/blob/master/jailkit-lib.pl#L299

`chown` does indeed update file **Change** time (i.e. `ctime`). Although I don't understand why Virtualmin doesn't rely on using **Modify** time (i.e. `mtime`) instead? It doesn't seem right to use `ctime` for this purpose. To be clear, using `chmod` doesn't change `mtime`, only `ctime`.

> you added this code recently .... do we really need that chmod?

Well, I added `-R` flag, to the [original code](https://github.com/virtualmin/virtualmin-gpl/commit/304991d9406cf819a35e9b15ce06b92257709e26), which would affect files under `~/virtualmin-backup`. Sadly, I don't remember why exactly I had to add `-R` flag but it couldn't be by accident.

**Comment by jcameron** [comment]
It's really the tar command that looks at those fields, not Virtualmin. 

I think we should remove that chmod command because I don't see why it's needed. Or maybe only do it when files are first copied?

**Comment by iliajie** [comment]
> It's really the tar command that looks at those fields, not Virtualmin.

I have just checked `backups-lib.pl` and we don't actually use `[10]` anywhere from the `stat`. It's always `[9]`, so you're right, it's not Virtualmin. However, `tar` also relies on `mtime`, right? So how do we get this issue just from running `chmod` over backups, which doesn't change `mtime`?

> I think we should remove that chmod command because I don't see why it's needed.

Well, could you try remembering why it was added in the [first place](https://github.com/virtualmin/virtualmin-gpl/commit/304991d9406cf819a35e9b15ce06b92257709e26)? It couldn't be by accident that we wanted to remove write permission for the group?

> Or maybe only do it when files are first copied?

What if files are added later?

**Comment by jcameron** [comment]
Hmm, looks like I added it to fix a problem with a specific directory. I think the right fix is to exclude /home from the chmod..

**Comment by iliajie** [comment]
Jamie, check [this PR](https://github.com/virtualmin/virtualmin-gpl/pull/763)
